src/logicblocks/event/processing/broker/broker/in_memory.py

```diff
@@ -28,13 +34,41 @@ def dependencies(self) -> EventBrokerDependencies:
             ),
             lock_manager=InMemoryLockManager(),
             event_source_factory=InMemoryEventStoreEventSourceFactory(
-                adapter=InMemoryEventStorageAdapter()
+                adapter=adaptor
             ),
         )
 
 
+@overload
+def make_in_memory_event_broker(
+    node_id: str,
+    settings: EventBrokerSettings,
+    *,
+    store: EventStore[InMemoryEventStorageAdapter],
+) -> EventBroker:
+    pass
+
+
+@overload
 def make_in_memory_event_broker(
     node_id: str,
     settings: EventBrokerSettings,
+    *,
+    adaptor: InMemoryEventStorageAdapter,
 ) -> EventBroker:
-    return InMemoryEventBrokerBuilder(node_id).prepare().build(settings)
+    pass
+
+
+def make_in_memory_event_broker(
```

Comment:
I don't think you should support and `adapter` and a `store`, just an `adapter` and if the broker internally needs a store, it creates one.

Reply:
They are stateless and there's nothing wrong with having more than one instance floating around.